Monika-After-Story/MonikaModDev
# Consumables Dialogue Props

Adds dialogue props dict as a property of `MASConsumable`s

These properties will be used to adjust the dialogue for generic labels to be more flexible for more consumables (prep for when we add more)

Current added props which are used:
- `mas_consumables.PROP_CONTAINER`
  - This is used in place of the old `container` property
- `mas_consumables.PROP_OBJ_REF`
  - This is used for consumables which don't necessarily have a container. For example cake, which would be referenced to as a *slice*
  - At the moment, this is only used if we don't have a container
- `mas_consumables.PROP_PLUR`
  - This is used to mark whether or not this consumable should be referenced to in plural or not

Also added some prefab `dlg_props` dicts to be used for consumables

Additionally, this also implements a maximum stock amount for consumables. This defaults to 150, but can be changed per consumable as it is a property which can be set on init.

- **NOTE:** If none of the props are present, there are fallback bits of dialogue to account for this. They may not always line up, but realistically, there shouldn't be a reason to not include at least one of the above reference properties (container/obj_ref).

# Testing
- Experiment with combinations of the dialogue properties and make sure that dialogue fits and is actually proper English
- Verify that there's no crashes
- Verify that you can only give up to a maximum of 150 servings of coffee/hotchoc (or if you adjust the value, that maximum)
- Verify that if you were over 150 servings of a consumable, the update script will adjust this and set you to the max stock amount
